src/binary_carrier.ts

```diff
@@ -5,7 +5,10 @@
  * this class is only one such type of object that can be used.
  */
 export default class BinaryCarrier {
-    constructor(binaryData) {
+
+    public buffer?: ArrayLike<number>;
+
+    constructor(public binaryData?: ArrayLike<number>) {
```

Comment:
Agree, but again, I just converted it to TypeScript. It is optional because there are test cases that call this without passing any parameter.